checks/burrowsc/burrow.go

```diff
@@ -0,0 +1,236 @@
+package burrowsc
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/hootsuite/healthchecks"
+)
+
+// Burrow consumer group and partition statuses (https://github.com/linkedin/Burrow/wiki/http-request-consumer-group-status)
+const (
+	OK       = "OK"
+	WARN     = "WARN"
+	ERR      = "ERR"
+	STOP     = "STOP"
+	STALL    = "STALL"
+	NOTFOUND = "NOTFOUND"
+)
+
+type LagResponse struct {
+	Error     bool      `json:"error"`
+	Message   string    `json:"message"`
+	LagStatus LagStatus `json:"status"`
+	Request   struct {
+		URL  string `json:"url"`
+		Host string `json:"host"`
+	} `json:"request"`
+}
+
+type LagStatus struct {
+	Cluster        string      `json:"cluster"`
+	Group          string      `json:"group"`
+	Status         string      `json:"status"`
+	Complete       float64     `json:"complete"`
+	Partitions     []Partition `json:"partitions"`
+	PartitionCount int         `json:"partition_count"`
+	MaxLag         Partition   `json:"maxlag"`
+	TotalLag       int64       `json:"totallag"`
+}
+
+type Partition struct {
+	Topic     string `json:"topic"`
+	Partition int    `json:"partition"`
+	Owner     string `json:"owner"`
+	ClientID  string `json:"client_id"`
+	Status    string `json:"status"`
+	Start     struct {
+		Offset    int64 `json:"offset"`
+		Timestamp int64 `json:"timestamp"`
+		Lag       int64 `json:"lag"`
+	} `json:"start"`
+	End struct {
+		Offset    int64 `json:"offset"`
+		Timestamp int64 `json:"timestamp"`
+		Lag       int64 `json:"lag"`
+	} `json:"end"`
+	CurrentLag int64   `json:"current_lag"`
+	Complete   float64 `json:"complete"`
+}
+
+type BurrowStatusChecker struct {
+	BaseUrl              string  // Base url of the Burrow API
+	Cluster              string  // The Kafka cluster to monitor
+	ConsumerGroup        string  // The consumer group on the cluster to monitor
+	Topic                *string // Optional if monitoring the status of a specific topic, leave nil to get the consumer group status
+	CriticalLagThreshold *int64  // Optional lag threshold to trigger a critical alert when exceeded
+}
+
+func (b BurrowStatusChecker) CheckStatus(name string) healthchecks.StatusList {
+	baseUrl := strings.TrimSuffix(b.BaseUrl, "/")
+	url := fmt.Sprintf("%s/%s/consumer/%s/lag", baseUrl, b.Cluster, b.ConsumerGroup)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return healthchecks.StatusList{
+			StatusList: []healthchecks.Status{
+				{
+					Description: name,
+					Result:      healthchecks.CRITICAL,
+					Details:     err.Error(),
+				},
+			},
+		}
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return healthchecks.StatusList{
+			StatusList: []healthchecks.Status{
+				{
+					Description: name,
+					Result:      healthchecks.CRITICAL,
+					Details:     err.Error(),
```

Comment:
do we do err.Error() or do we return a failed string in other place? (i.e just read "Failed to check burrow status" etc etc

Reply:
I copied this over from the `HttpStatusChecker`, it looks like we do `err.Error()` in cases where we have an `error` available in the code.

Reply:
Ok 👍